packages/astro/src/core/render/core.ts

```diff
@@ -146,7 +144,17 @@ export async function render(
 		ssr,
 	});
 
-	let page = await renderPage(result, Component, pageProps, null);
+	let page: Awaited<ReturnType<typeof renderPage>>;
+	if (!Component.isAstroComponentFactory) {
+		const props: Record<string, any> = { ...(pageProps ?? {}), 'server:root': true };
+		const html = await renderComponent(result, Component.name, Component, props, null);
+		page = {
+			type: 'html',
+			html: html.toString()
+		}
```

Comment:
If the module exports a non-astro file, we pass it through our component pipeline instead, which will use any provided renderers to render the component